models/migrations/v96.go

```diff
@@ -47,12 +50,14 @@ func deleteOrphanedAttachments(x *xorm.Engine) error {
 		for _, attachment := range attachements {
 			ids = append(ids, attachment.ID)
 		}
-		if _, err := sess.In("id", ids).Delete(new(Attachment)); err != nil {
-			return err
+		if len(ids) > 0 {
+			if _, err := sess.In("id", ids).Delete(new(Attachment)); err != nil {
+				return err
+			}
 		}
 
 		for _, attachment := range attachements {
-			if err := os.RemoveAll(models.AttachmentLocalPath(attachment.UUID)); err != nil {
+			if err := storage.Attachments.Delete(relativePath(attachment.UUID)); err != nil {
```

Comment:
This should make sure `storage` is always local, as other options don't exist before this v96.go is executed.

Reply:
But how did we know where the files stored?

Reply:
This is to delete broken attachments whatever which the attachments stored.

Reply:
This is old migration code (1.10.0), so I think of the following scenario:

- User has 1.9.5.
- Installs the 1.13.x binary <-- doesn't run it yet.
- Edits the attachment storage options in `app.ini` to use S3.
- The user now runs the 1.13.x binary for the first time. All the migrations are run, including v96. But v96 runs against the S3 storage which is wrong because the attachments are still in local storage.

That's why I think v96 should always run against local storage disregarding whatever is specified in `app.ini`, or it should at least fail if the storage is not `local`.

Reply:
v96.go is developed on an old version gitea, but it maybe executed after this PR merged. So we shouldn't assume it's always the local path.

Reply:
@lunny can the user have attachments on a remote storage _before_ running v96.go? I don't think they can. 🤔

Reply:
I think you are right but he has missed the origial config(he changed). So v96.go cannot delete the right files if we use local storage.

Reply:
`gitea migrate-storage` will cause a db update - this code has to use the local storage variant.